### PR TITLE
Added Solutions to Interview Questions

### PR DESCRIPTION
**Pull request by Akshay-jain22**

### CODE REVIEW

**Comment by Akshay-jain22** [comment]
Link to the Markdown : https://github.com/Akshay-jain22/IRIS-RoR-Bootcamp-2020/blob/session_6_akshay_jain/interview_questions.md

**Comment by Akshay-jain22** [comment]
Tell if some of the solutions are not correct. It'll be helpful :)

**Comment by abhishekkumar2718** [comment]
> Link to the Markdown : https://github.com/Akshay-jain22/IRIS-RoR-Bootcamp-2020/blob/session_6_akshay_jain/interview_questions.md

1. Pull from the repository, create a new branch and push again 
2. Style the answers as:

<details><summary>What is meant by frontend and backend</summary>
<p>


The part of a website that user interacts with directly is termed as front end. It is also referred to as the 'client side' of the application.

Backend is 'server side' of the website. It stores and arranges data, and also makes sure everything on the client-side of the website works fine. It is the part of the website that you cannot see and interact with.
</p>
</details>

3. Add external links to resources on the topic the question is based on.

File: interview_questions.md [hunk 1]
```diff
@@ -11,127 +11,916 @@ and Rails, grouped by category.
...
+  ```
+
 - What is meant by responsive web design?
+  ```
+  A responsive website has a fluid and flexible layout which adjusts according to screen size. Basically, your website will look great and work well on a laptop, a tablet, and a mobile phone's browser.
+  ```
+
 - What are some common backend web frameworks?
+  ```
+  Backend Frameworks : Laravel Framework, Flask Framework, Django, etc.
```
Comment (abhishekkumar2718) [inline interview_questions.md:36]: Woah, no Rails?

File: interview_questions.md [hunk 2]
```diff
@@ -11,127 +11,916 @@ and Rails, grouped by category.
...
+  ```
+
 - Explain MVC architecture and its benefits?
+  ```
+  The Model-View-Controller (MVC) is an architectural pattern that separates an application into three main logical components: the model, the view, and the controller.
```
Comment (abhishekkumar2718) [inline interview_questions.md:41]: You should define what is meant by the terms model, view and controller further.

File: interview_questions.md [hunk 3]
```diff
@@ -11,127 +11,916 @@ and Rails, grouped by category.
...
+  - Easier to Debug as we have multiple levels properly written in the application.
+  - The Modification Never Affects The Entire Model.
+  - Faster Web Application Development Process
+  ```
+
 - What are common ways of storing user information?
+  ```
+  Common ways of storing user information :-
+  - Data Storage on the Cloud
+  - Storing Your Data Locally
```
Comment (abhishekkumar2718) [inline interview_questions.md:53]: You can write `using database applications locally`

File: interview_questions.md [hunk 4]
```diff
@@ -11,127 +11,916 @@ and Rails, grouped by category.
...
+
+  Iterating over hash :-
+  hash.each do |key, value|
+    # Some Code
+  end
+  ```
+
 - What are some differences between `nil` and `false`?
+  ```
+  nil means no data value but memory is still allocated for null.
```
Comment (abhishekkumar2718) [inline interview_questions.md:377]: This is not the database value `null` but the ruby object `nil` - incorrect answer and refer to this: https://learn.co/lessons/truthiness-in-ruby-readme and https://gist.github.com/jfarmer/2647362

**Review by abhishekkumar2718 — CHANGES_REQUESTED** [review]

File: interview_questions.md (hunk 2 above)
Comment (Akshay-jain22) [reply interview_questions.md:41]: Ohk👍

File: interview_questions.md (hunk 4 above)
Comment (Akshay-jain22) [reply interview_questions.md:377]: Ig it got auto-correct from nil to null 😄..ill correct that

**Comment by Akshay-jain22** [comment]
> > Link to the Markdown : https://github.com/Akshay-jain22/IRIS-RoR-Bootcamp-2020/blob/session_6_akshay_jain/interview_questions.md
> 
> Thanks for taking on this issue - it will take a lot of time to carefully review and polish the pull request, so be patient between round of reviews.
> 
> 1. Pull from the repository, create a new branch and push again
> 2. Style the answers as:
> 
> What is meant by frontend and backend
> 1. Add external links to resources on the topic the question is based on.

It's on new session 6 branch only.. Should I create a interview questions branch?

**Comment by abhishekkumar2718** [comment]
> > > Link to the Markdown : https://github.com/Akshay-jain22/IRIS-RoR-Bootcamp-2020/blob/session_6_akshay_jain/interview_questions.md
> > 
> > 
> > Thanks for taking on this issue - it will take a lot of time to carefully review and polish the pull request, so be patient between round of reviews.
> > 
> > 1. Pull from the repository, create a new branch and push again
> > 2. Style the answers as:
> > 
> > What is meant by frontend and backend
> > 
> > 1. Add external links to resources on the topic the question is based on.
> 
> It's on new session 6 branch only.. Should I create a interview questions branch?

Yes, session 6 is not the same as interview questions

**Comment by Akshay-jain22** [comment]
Closing this PR..Check new PR for any suggestions

**Pull request closed without merging**